ServantHDBC/src/Api/Authentication.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Api.Authentication (AuthAPI, authHandlers) where
+
+import Control.Monad.Trans.Except
+import Servant
+
+type Auth = String
+type AuthAPI = "sessions" :>
+                         ("new" :> Post '[JSON] Auth
+                          :<|> "refresh" :> Post '[JSON] Auth
+                          :<|> "destroy" :> Post '[JSON] Auth)
+
+authNew :: ExceptT ServantErr IO Auth
```

Comment:
@saurabhnanda `ServantErr` is record with 4 fields... more details here 
https://hackage.haskell.org/package/servant-server-0.9.0.1/docs/Servant-Server-Internal-ServantErr.html#t:ServantErr

Yes, you need to use `ExceptT` as that what Servant is using or expecting (https://github.com/haskell-servant/servant/blob/master/servant-server/src/Servant/Server/Internal/ServantErr.hs#L22)

You can have `Header` in the Type... `Header "something" a` cookie is also part of header so I guess it can be in the type.